File: clients/typescript/examples/tables-alter-type-add.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get a database
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});

// Drop a user-defined type
(async function () {
  await database.alterType("member", {
    operation: {
      add: {
        fields: {
          email: "text",
          credits: "int",
        },
      },
    },
  });
})();

File: clients/python/examples/tables-alter-type-add.py
from astrapy import DataAPIClient
from astrapy.info import (
    AlterTypeAddFields,
    ColumnType,
    TableScalarColumnTypeDescriptor,
)

# Get a database
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)

# Add fields to a user-defined type
database.alter_type(
    "member",
    AlterTypeAddFields(
        fields={
            "email": TableScalarColumnTypeDescriptor(ColumnType.TEXT),
            "credits": TableScalarColumnTypeDescriptor(ColumnType.INT),
        }
    ),
)
